src/connector/src/source/cdc/external/mysql.rs

```diff
@@ -375,10 +374,15 @@ impl ExternalTableReader for MySqlExternalTableReader {
     ) -> BoxStream<'_, ConnectorResult<OwnedRow>> {
         self.snapshot_read_inner(table_name, start_pk, primary_keys, limit)
     }
+
+    async fn disconnect(&self) -> ConnectorResult<()> {
```

Comment:
You can pass ownership of self here so that you don't need to clone the pool: `async fn disconnect(self)`

Reply:
A bit hard because we need to `drop(upstream_table_reader);` after `upstream_table_reader.disconnect()`, since other cdc's table_reader need manually drop.